### PR TITLE
goose 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by mkozjak**
Signed-off-by: Mario Kozjak <kozjakm1@gmail.com>

- [✓] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [✓] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [✓] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [✓] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

```goose``` is a Golang command line interface tool for the https://github.com/pressly/goose library.

It pulls source from the stable release tarball, uses brew go to install from source, and installs the migrate binary to /usr/local/bin/ via formula's ```bin.install``` function.

Resolves pressly/goose#42.

### CODE REVIEW

**Review by ilovezfs — CHANGES_REQUESTED** [review]
In new formulae, we generally want to avoid `go_resource` blocks. Please file an issue upstream requesting use of a Go vendor directory or one of the various Go vendoring tools.

**Comment by mkozjak** [comment]
@ilovezfs: I guess a good example is the 'hugo' formula? I believe the dependency should then have a 'vendor' directory with a vendor.json file in it when the vendoring tool is govendor...

**Comment by ilovezfs** [comment]
`flint-checker` is a good example for the basic Go vendor directory. Take a look at https://github.com/pengwynn/flint/issues/46.

**Comment by mkozjak** [comment]
Created an issue. Let's hope for the best. :)

**Comment by ilovezfs** [comment]
Thanks @mkozjak!

**Comment by mkozjak** [comment]
@ilovezfs Somehow this got closed while I was syncing with upstream. Can you, please, reopen? Or should I create a new issue?

**Comment by ilovezfs** [comment]
It got closed because it had 0 diff to master.

**Comment by mkozjak** [comment]
Thanks! Now we only wait for ```dep```.

**Comment by ilovezfs** [comment]
```
require "language/go"

class Goose < Formula
  desc "Go Language's command-line interface for database migrations"
  homepage "https://github.com/pressly/goose"
  url "https://github.com/pressly/goose/archive/v1.0.tar.gz"
  sha256 "d1b84411dcb5abc211d52de65f1eeaa9ea4a1eeaee0400b3380d9d24e1dc8b1b"
  head "https://github.com/pressly/goose.git"

  depends_on "go" => :build

  go_resource "github.com/golang/dep" do
    url "https://github.com/golang/dep.git",
        :revision => "c79b048e07eccf76d323d4c2e88d7c6d72ea735f"
  end

  def install
    ENV["GOPATH"] = buildpath

    (buildpath/"src/github.com/pressly/goose").install buildpath.children
    Language::Go.stage_deps resources, buildpath/"src"

    cd "src/github.com/golang/dep" do
      system "go", "install", "github.com/golang/dep/cmd/dep"
    end
    ENV.append_path "PATH", buildpath/"bin"

    cd "src/github.com/pressly/goose" do
      system "dep", "ensure"
      system "go", "build", "-o", "goose", "github.com/pressly/goose/cmd/goose"
      bin.install "goose"
    end
  end

  test do
    system bin/"goose", "sqlite3", "./foo.db", "status", "create"
  end
end
```

**Comment by ilovezfs** [comment]
But it only works for `--HEAD`

**Comment by ilovezfs** [comment]
And this gets it using the Homebrew cache if we want to pursue that at some point.
```
    ENV["GOPATH"] = buildpath
    go_cache = HOMEBREW_CACHE/"go_cache/#{name}/dep"
    go_cache.mkpath

    (buildpath/"src/github.com/pressly/goose").install buildpath.children
    Language::Go.stage_deps resources, buildpath/"src"

    (buildpath/"pkg").install_symlink go_cache
```

**Comment by mkozjak** [comment]
Should I compile it all now and push?

**Comment by ilovezfs** [comment]
@mkozjak let's see whether upstream is willing to create a new tag now or not so we know how to proceed.

**Comment by VojtechVitek** [comment]
@ilovezfs @mkozjak I will be releasing v2.0 later this week. Can you hold off for a bit?

**Comment by ilovezfs** [comment]
@VojtechVitek not a problem. We've got it working with `dep`: https://github.com/Homebrew/homebrew-core/pull/14724#issuecomment-309815183

So whenever you're ready.

**Comment by mkozjak** [comment]
@ilovezfs Goose v2.0.0 is here. Should we proceed? Pushed formula updates which use ```dep``` (without caching support for now).

File: Formula/goose.rb [hunk 1]
```diff
@@ -3,21 +3,30 @@
...
     Language::Go.stage_deps resources, buildpath/"src"
-    (buildpath/"src/github.com/pressly").mkpath
-    ln_s buildpath, buildpath/"src/github.com/pressly/goose"
 
-    cd buildpath/"cmd/goose" do
-      system "go", "build", "-o", "./goose", "."
+    cd "src/github.com/golang/dep" do
+      system "go", "install", "github.com/golang/dep/cmd/dep"
+    end
+    ENV.append_path "PATH", buildpath/"bin"
```
Comment (dunn) [inline Formula/goose.rb]: Instead of changing the path, below just do `system buildpath/"bin/dep", "ensure"`.

Comment (mkozjak) [inline Formula/goose.rb]: Done!

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @mkozjak! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by mkozjak** [comment]
@ilovezfs And thank you for giving so much time, effort and positive energy in such an amazing project like this one!

**Comment by VojtechVitek** [comment]
Well done guys!

**Pull request closed without merging**